zerver/tests/test_auth_backends.py

```diff
@@ -1053,27 +1061,62 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
         else:
             self.assertIn('do_confirm/' + confirmation_key, result.url)
             do_confirm_url = result.url
+
+        if user_avatar_url:
+            # If there is an avatar_url, then there must be a user_avatar_url
+            # on the preregistration user.
+            prereg_user = confirmation.content_object
+            self.assertEqual(prereg_user.user_avatar_url, user_avatar_url)
+
         result = self.client_get(do_confirm_url, name = name)
         self.assert_in_response('action="/accounts/register/"', result)
         data = {"from_confirmation": "1",
                 "key": confirmation_key}
+
         result = self.client_post('/accounts/register/', data)
+        expected_social_avatar_hash = ''
         if not skip_registration_form:
             self.assert_in_response("We just need you to do one last thing", result)
 
             # Verify that the user is asked for name but not password
             self.assert_not_in_success_response(['id_password'], result)
             self.assert_in_success_response(['id_full_name'], result)
+            if user_avatar_url:
+                # Verify if there is a checkbox for the user choose to use their
+                # social avatar.
+                self.assert_in_success_response(['use_social_avatar'], result)
             # Verify the name field gets correctly pre-populated:
             self.assert_in_success_response([expected_final_name], result)
 
-            # Click confirm registration button.
-            result = self.client_post(
-                '/accounts/register/',
-                {'full_name': expected_final_name,
-                 'key': confirmation_key,
-                 'terms': True})
-
+            with responses.RequestsMock(assert_all_requests_are_fired=True) as mock_upload:
+                if use_social_avatar:
+                    expected_social_avatar_content = open(expected_social_avatar_content_path, 'rb').read()
+                    if not avatar_upload_error:
+                        mock_upload.add(
+                            method='GET',
+                            url=user_avatar_url,
+                            status=200,
+                            content_type='binary',
+                            body=expected_social_avatar_content
+                        )
+
+                    elif avatar_upload_error:
+                        mock_upload.add(
+                            method='GET',
+                            url=user_avatar_url,
+                            status=302,
```

Comment:
why specifically 302? Wouldn't 500 or something like that be better? (I guess it doesn't matter much, but `302` is a bit of a surprising value to put)

Reply:
I've put an URL that doesn't exist to see the response and that was the response status, which now that I'm thinking about it is a bit odd indeed. Gonna change to 500.